File: 0324-wiggle-sort-ii/0324-wiggle-sort-ii.py
class Solution:
    def wiggleSort(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        increasing = nums.copy()
        increasing.sort()
        left = (len(nums)-1)//2
        right = len(nums)-1
        for i in range(len(nums)):
            if i%2 == 0:
                nums[i] = increasing[left]
                left -= 1
            else:
                nums[i] = increasing[right]
                right -= 1

File: 0324-wiggle-sort-ii/0324-wiggle-sort-ii.go
import (
    "sort"
)

func wiggleSort(nums []int)  {
    sorted := make([]int, len(nums))
    copy(sorted, nums)
    sort.Ints(sorted)
    
    left := (len(sorted) - 1) / 2
    right := len(sorted) - 1
    
    for i := 0; i < len(sorted); i++ {
        if i % 2 == 0 {
            nums[i] = sorted[left]
            left--
        } else {
            nums[i] = sorted[right]
            right--
        }
    }
}